### PR TITLE
Add EditorFeatures.Cocoa

### PR DESCRIPTION
**Pull request by tmat**
Matches commit https://devdiv.visualstudio.com/DevDiv/_git/VS-Platform/commit/10204c06cc105f2f2455ae6f84f34eac2ce21c93

Corresponding change in VS-Platform: https://devdiv.visualstudio.com/DevDiv/_git/VS-Platform/pullrequest/283665

### CODE REVIEW

**Comment by KirillOsenkov** [comment]
@DavidKarlas @Therzok @davidwengier @gundermanc @olegtk FYI

File: src/EditorFeatures/Core.Cocoa/InlineRename/CommandHandlers/RenameCommandHandler_RenameHandler.cs [hunk 1]
```diff
@@ -35,7 +39,7 @@ public VSCommanding.CommandState GetCommandState(RenameCommandArgs args)
                 return VSCommanding.CommandState.Unspecified;
             }
 
-            var documents = textContainer.GetRelatedDocuments();
+            _ = textContainer.GetRelatedDocuments();
```
Comment (sharwell) [inline src/EditorFeatures/Core.Cocoa/InlineRename/CommandHandlers/RenameCommandHandler_RenameHandler.cs:40]: 💭 A follow-up issue to remove the items getting assigned to discards would be good.

File: src/EditorFeatures/Core.Cocoa/Microsoft.CodeAnalysis.EditorFeatures.Cocoa.csproj [hunk 2]
```diff
@@ -30,4 +30,8 @@
     <PackageReference Include="XamarinMac" Version="$(XamarinMacVersion)" />
     <PackageReference Include="System.ValueTuple" Version="$(SystemValueTupleVersion)" />
   </ItemGroup>
+  <ItemGroup>
+    <PublicAPI Include="PublicAPI.Shipped.txt" />
+    <PublicAPI Include="PublicAPI.Unshipped.txt" />
```
Comment (sharwell) [inline src/EditorFeatures/Core.Cocoa/Microsoft.CodeAnalysis.EditorFeatures.Cocoa.csproj:35]: 💭  I thought this was done in Directory.Build.props

File: src/EditorFeatures/Core.Cocoa/NavigationCommandHandlers/FindBaseSymbolsCommandHandler.cs [hunk 3]
```diff
@@ -86,12 +69,14 @@ protected override bool TryExecuteCommand(int caretPosition, Document document,
 
                     using (Logger.LogBlock(
                         FunctionId.CommandHandler_FindAllReference,
-                        KeyValueLogMessage.Create(LogType.UserAction, m => m ["type"] = "streaming"),
+                        KeyValueLogMessage.Create(LogType.UserAction, m => m["type"] = "streaming"),
                         context.CancellationToken))
                     {
                         try
                         {
+#pragma warning disable CA2007 // Consider calling ConfigureAwait on the awaited task
```
Comment (sharwell) [inline src/EditorFeatures/Core.Cocoa/NavigationCommandHandlers/FindBaseSymbolsCommandHandler.cs:77]: 📝 These are fine, but we should have one dedicated follow-up issue to address all CA2007 suppressions in this project.

**Review by sharwell — COMMENTED** [review]
⚠️ The PublicAPI files appear to be added as binary instead of text. They should be re-saved as utf-8-bom.

File: src/EditorFeatures/Core.Cocoa/InlineRename/CommandHandlers/RenameCommandHandler_RenameHandler.cs (hunk 1 above)
Comment (tmat) [reply src/EditorFeatures/Core.Cocoa/InlineRename/CommandHandlers/RenameCommandHandler_RenameHandler.cs:40]: Definitely. Already filed https://github.com/dotnet/roslyn/issues/48871 to review all suppressions. I'll add a note to revisit discards too.

File: src/EditorFeatures/Core.Cocoa/NavigationCommandHandlers/FindBaseSymbolsCommandHandler.cs (hunk 3 above)
Comment (tmat) [reply src/EditorFeatures/Core.Cocoa/NavigationCommandHandlers/FindBaseSymbolsCommandHandler.cs:77]: Yup: #48871 to review all suppressions.

File: src/EditorFeatures/Core.Cocoa/Microsoft.CodeAnalysis.EditorFeatures.Cocoa.csproj (hunk 2 above)
Comment (tmat) [reply src/EditorFeatures/Core.Cocoa/Microsoft.CodeAnalysis.EditorFeatures.Cocoa.csproj:35]: Other projects list them explicitly in the project file.

**Review by sharwell — APPROVED** [review]

**Comment by jasonmalinowski** [comment]
@tmat Just to confirm, I see there's a bunch of files that seem cloned in this PR. I presume the sequence of events was:

1. The files used to live in the Wpf layer.
2. It got cloned into the Cocoa layer in the past rather than refactoring.
3. We're migrating that repository verbatim into our repo.
4. Now that it's here we can cleanup/deduplicate as appropriate.

Is that correct?

**Comment by KirillOsenkov** [comment]
Yes, Cocoa was a fork of EditorFeatures.Wpf which first lived in the VSMac repo, then got moved to the VS-Platform repo (for expediency) and now found it's best home in the Roslyn repo.

**Pull request merged**